backend/devices/bitbox/device.go

```diff
@@ -1214,48 +1237,72 @@ func (dbb *Device) ECDHchallenge() error {
 
 // StartPairing creates, stores and returns a new channel and finishes the pairing asynchronously.
 func (dbb *Device) StartPairing() (*relay.Channel, error) {
+	var removed bool
+	dbb.mu.Lock()
 	if dbb.channel != nil {
-		if err := dbb.channel.RemoveConfigFile(); err != nil {
+		if err := dbb.channel.RemoveConfigFile(dbb.channelConfigDir); err != nil {
+			dbb.mu.Unlock()
 			return nil, errp.WithStack(err)
 		}
 		dbb.channel = nil
+		removed = true
+	}
+	dbb.mu.Unlock()
+	if removed {
 		dbb.fireEvent("pairingFalse", nil)
 	}
+
 	channel := relay.NewChannelWithRandomKey()
 	go dbb.processPairing(channel)
 	return channel, nil
 }
 
 // Paired returns whether a channel to a mobile exists.
 func (dbb *Device) Paired() bool {
-	return dbb.channel != nil
+	return dbb.mobileChannel() != nil
 }
 
 // PingMobile pings the mobile if a channel exists. Only returns no error if the pong was received.
 func (dbb *Device) PingMobile() error {
-	if dbb.channel != nil {
-		if err := dbb.channel.SendPing(); err != nil {
-			return err
+	mob := dbb.mobileChannel()
+	if mob == nil {
+		return errp.New("bitbox: device's mobile channel is nil")
```

Comment:
But this isn't a programming error. Because there are separate goroutines which modify the `dbb.channel` field, it is possible in theory, even if looking closing at the code today it doesn't seem so.

I only added this because `PingMobile` is exported. So, it might be that the Device's goroutines are updating its channel field internally while an external caller calls PingMobile.

Reply:
Always happy to unexport. Let me do it in a separate pull request after this is merged.